### PR TITLE
Let FreeRotator start with a specific angle index

### PR DESCRIPTION
**Pull request by Cattlesquat**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/counters/FreeRotator.java [hunk 1]
```diff
@@ -981,12 +1001,13 @@ public String getType() {
...
       se.append(nameConfig.getValueString());
       se.append(descConfig.getValueString());
+      se.append(startAngleConfig.getValueString());
       return ID + se.getValue();
     }
 
     @Override
     public String getState() {
-      return "0";
+      return startAngleConfig.getValueString();
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/FreeRotator.java:1010]: return rotator.getState();

File: vassal-app/src/main/java/VASSAL/counters/FreeRotator.java [hunk 2]
```diff
@@ -859,6 +873,7 @@ public boolean testEquals(Object o) {
     private final StringConfigurer nameConfig;
     private final StringConfigurer descConfig;
     private final JLabel anyLabel;
+    private final IntConfigurer startAngleConfig;
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/FreeRotator.java:876]: private final FreeRotator rotator;

File: vassal-app/src/main/java/VASSAL/counters/FreeRotator.java [hunk 3]
```diff
@@ -931,6 +946,11 @@ public Ed(FreeRotator p) {
       rndKeyConfig = new NamedHotKeyConfigurer(p.rotateRNDKey);
       panel.add(rndKeyConfig.getControls(), "wrap"); // NON-NLS
 
+      final JLabel startAngleLabel = new JLabel(Resources.getString("Editor.FreeRotator.start_angle"));
+      panel.add(startAngleLabel);
+      startAngleConfig = new IntConfigurer(p.startAngleIndex);
+      panel.add(startAngleConfig.getControls());
+
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/FreeRotator.java:953]: rotator = p;

**Review by BrentEaston — COMMENTED** [review]
I don;t think this is needed. Can't you just fix the bug in FreeRotator.Ed.getState() (see below) and then rotate the unit in the palette?

File: vassal-app/src/main/java/VASSAL/counters/FreeRotator.java (hunk 1 above)
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/FreeRotator.java:1010]: Ah, I know why they returned "0". For my solution, you would need to check if the type has changed from fixed to free or v/v and convert the starting state if it is non-zero.

**Comment by Cattlesquat** [comment]
Okay, so "good news and bad news":

GOOD NEWS - with a few extra details worked out (see current version of PR) this solution now "works"

BAD NEWS - After some hair-pulling thinking there was some weird bug in my code, I have discovered that actually there is a crash bug IN THE ORIGINAL CODE. This PR doesn't exacerbate it, nor does it fix it, but *testing* this PR reveals the original crash bug.

REPRO STEPS (original code, don't need to use this PR):
1. Make a piece with a FreeRotator w/ e.g. 8 valid angle facings (I assume the default 6 works too, I was using 8)
2. Manually rotate the image a step (i.e. modifying the state)
3. Now open the FreeRotator (Can Rotate) trait to edit
4. Check the checkbox to "allow arbitrary facings"
5. Click OK button

EXPECTED: yay you edited your thingy and we cleared the state back to 0
ACTUAL: crash

The crash is somewhere in updating the image/sizing-of-image/bounding-box-of-the-sizing code for redrawing the thing, and has apparently been there for Who Knows How Long. 

Obviously the case where we have a pre-rotated state of a thing and then someone decides to change to an arbitrary facing is so unusual/rare that we've never noticed it until now. 

So I propose mainly that we merge this and move on, since it handles the key issues at hand correctly, enables my use case, and doesn't make the existing bug worse. Fixing the revealed bug is probably your wheelhouse as far as configurers-interacting-with-things; I'm not sure it's even an "important" bug, we might just want to open an issue and leave it for now?

File: vassal-app/src/main/resources/VASSAL/i18n/Editor.properties [hunk 4]
```diff
@@ -675,6 +675,7 @@ Editor.FreeRotator.rotate_ccw_command_description=Rotate CCW command
 Editor.FreeRotator.rotate_random_command_description=Rotate Random command
 Editor.FreeRotator.allow_arbitrary_rotations=Allow arbitrary rotations
 Editor.FreeRotator.number_of_allowed_facings=Number of allowed facings
+Editor.FreeRotator.start_angle=Starting angle index (or angle if arbitrary rotations allowed)
```
Comment (BrentEaston) [inline vassal-app/src/main/resources/VASSAL/i18n/Editor.properties:678]: No longer needed?

File: vassal-app/src/main/java/VASSAL/counters/FreeRotator.java [hunk 5]
```diff
@@ -126,7 +126,7 @@
 
   public FreeRotator() {
     // modified for random rotation (added two ; )
-    this(ID + "6;];[;" + Resources.getString("Editor.FreeRotator.default_rotate_cw_command") + ";" + Resources.getString("Editor.FreeRotator.default_rotate_ccw_command") + ";;;;", null); //$NON-NLS-1$//
+    this(ID + "6;];[;" + Resources.getString("Editor.FreeRotator.default_rotate_cw_command") + ";" + Resources.getString("Editor.FreeRotator.default_rotate_ccw_command") + ";;;;;", null); //$NON-NLS-1$//
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/FreeRotator.java:129]: The extra ';' should no longer be needed as we are no longer adding the start  angle option?

**Review by BrentEaston — CHANGES_REQUESTED** [review]
Still some cruft left from the original solution.

**Comment by Cattlesquat** [comment]
Cruft removed!

**Review by BrentEaston — APPROVED** [review]

**Pull request merged**